syntax = "proto3";

package connectfour;

enum Mark {
  MARK_RED = 0;
  MARK_YELLOW = 1;
}

message Game {
  int32 id = 1;
  repeated Move moves = 2;
  Mark turn = 3;
  bool is_finished = 4;
  optional Mark winner = 5;
}

message CreateGameRequest {}

message GetGameRequest {
  int32 game_id = 1;
}

message MakeMoveRequest {
  int32 game_id = 1;
  Move move = 2;
}

message Move {
  Mark mark = 1;
  int32 column = 2;
}

service ConnectFour {
  rpc CreateGame (CreateGameRequest) returns (Game);
  rpc GetGame (GetGameRequest) returns (Game);
  rpc MakeMove (MakeMoveRequest) returns (Game);
}
